pandas/core/arrays/masked.py

```diff
@@ -1088,25 +1092,62 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
 
         # median, skew, kurt, sem
         op = getattr(nanops, f"nan{name}")
-        result = op(data, axis=0, skipna=skipna, mask=mask, **kwargs)
-
+        axis = kwargs.pop("axis", None)
+        result = op(data, axis=axis, skipna=skipna, mask=mask, **kwargs)
         if np.isnan(result):
-            return libmissing.NA
+            result = libmissing.NA
 
-        return result
+        return self._wrap_reduction_result(
+            name, result, skipna=skipna, axis=axis, **kwargs
+        )
+
+    def _reduce_and_wrap(self, name: str, *, skipna: bool = True, kwargs):
+        df = self.reshape(-1, 1)
```

Comment:
Hmm, that could well be right. Let me look into it and I'll get back.

Reply:
The comment wrapping in github is started to get to the point where I' sure if I've answered everything, But I've made a new version that uses 1d arrays like suggested here, it works the same and is simpler because we avoid possibly complexity of 2d arrays. There is still some 2d code improvements is this PR, but IMO we should keep it in because it makes the the 2d ops better.